server/tests/src/test/java/org/infinispan/server/persistence/PersistenceIT.java

```diff
@@ -21,16 +39,64 @@
 })
 public class PersistenceIT {
 
-   public static final DatabaseServerListener DATABASE = new DatabaseServerListener("h2", "mysql", "postgres");
+   private static final String DATABASES = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_TYPES);
+   private static final String DATABASE_LIBS = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_LIBS);
+   private static final Integer ARTIFACT_ID_POSITION = 1;
+   private static final String EXTERNAL_JDBC_DRIVER = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_EXTERNAL_DRIVERS);
+   private static String jdbcDriverFromFile = System.getProperty(TestSystemPropertyNames.INFINISPAN_TEST_CONTAINER_DATABASE_DRIVERS_FILE);
+
+   public static final DatabaseServerListener DATABASE_LISTENER = new DatabaseServerListener(getDatabases());
 
    @ClassRule
    public static InfinispanServerRule SERVERS =
          InfinispanServerRuleBuilder.config(System.getProperty(PersistenceIT.class.getName(), "configuration/PersistenceTest.xml"))
                                     .numServers(1)
                                     .runMode(ServerRunMode.CONTAINER)
-                                    .mavenArtifacts("com.h2database:h2:1.4.199",
-                                                    "mysql:mysql-connector-java:8.0.17",
-                                                    "org.postgresql:postgresql:jar:42.2.8")
-                                    .addListener(DATABASE)
+                                    .mavenArtifacts(getJdbcDrivers())
+                                    .artifacts(getJavaArchive())
+                                    .addListener(DATABASE_LISTENER)
                                     .build();
+
+   private static String[] getJdbcDrivers() {
+      jdbcDriverFromFile = jdbcDriverFromFile == null ? "src/test/resources/database/jdbc-drivers.txt" : jdbcDriverFromFile;
+      Map<String, String> jdbcDrivers = Exceptions.unchecked(() -> Files.lines(Paths.get(jdbcDriverFromFile))
+              .collect(Collectors.toMap(PersistenceIT::getArtifactId, Function.identity())));
+
+      // default jdbc drivers can also be set through system properties
+      // jdbc drivers can be updated or added
+      if(DATABASE_LIBS != null) {
+         Arrays.stream(DATABASE_LIBS.split(",")).forEach(it -> jdbcDrivers.put(getArtifactId(it), it));
+      }
```

Comment:
Here you could do `System.getProperties().computeIfPresent(..)` and remove the need for the DATABASE_LIBS var.